The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13647886445/job/38150018501) [(plain)](https://github.com/rust-lang/rust/commit/60c5855346576dea6666adb9e00479156242b495/checks/38150018501/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04
ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
  make \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 2.887 Building wheels for collected packages: reuse
#12 2.888   Building wheel for reuse (pyproject.toml): started
#12 3.100   Building wheel for reuse (pyproject.toml): finished with status 'done'
#12 3.101   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=0c2fd2aaf7b0bf8d6e131220aff14712a774c2ca462f3204d25460cbcf610b63
#12 3.101   Stored in directory: /tmp/pip-ephem-wheel-cache-9v9jbc71/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#12 3.104 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#12 3.509 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#12 3.509 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#12 4.051 Collecting virtualenv
#12 4.051 Collecting virtualenv
#12 4.108   Downloading virtualenv-20.29.2-py3-none-any.whl (4.3 MB)
#12 4.174      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 67.5 MB/s eta 0:00:00
#12 4.235 Collecting filelock<4,>=3.12.2
#12 4.243   Downloading filelock-3.17.0-py3-none-any.whl (16 kB)
#12 4.264 Collecting distlib<1,>=0.3.7
#12 4.272   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.279      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 88.0 MB/s eta 0:00:00
#12 4.316 Collecting platformdirs<5,>=3.9.1
#12 4.323   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#12 4.406 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.591 Successfully installed distlib-0.3.9 filelock-3.17.0 platformdirs-4.3.6 virtualenv-20.29.2
#12 DONE 4.7s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
DirectMap4k:      135104 kB
DirectMap2M:     6156288 kB
DirectMap1G:    12582912 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
WARN: `rust.debug-assertions = true` will prevent downloading CI rustc as alt CI rustc is not currently built with debug assertions.
---
fmt check
fmt: checked 5871 files
tidy check
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/src/librustdoc/html/static/js/search.js:1587: tab character
##[error]tidy error: /checkout/src/librustdoc/html/static/js/search.js:1588: tab character
##[error]tidy error: /checkout/src/librustdoc/html/static/js/search.js:1589: tab character
##[error]tidy error: /checkout/src/librustdoc/html/static/js/search.js:1593: tab character
##[error]tidy error: /checkout/src/librustdoc/html/static/js/search.js:1594: tab character
##[error]tidy error: /checkout/src/librustdoc/html/static/js/search.js:1597: tab character
##[error]tidy error: /checkout/src/librustdoc/html/static/js/search.js:1597: trailing whitespace
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'venv'
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'virtualenv'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (25.0.1)
All checks passed!
checking python file formatting
27 files already formatted
checking C++ file formatting
checking C++ file formatting
some tidy checks failed
Command has failed. Rerun with -v to see more details.
  local time: Tue Mar  4 06:56:53 UTC 2025
  network time: Tue, 04 Mar 2025 06:56:53 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/librustdoc/html/static/js/search.js
+++ b/src/librustdoc/html/static/js/search.js
@@ -2398,9 +2398,9 @@ class DocSearch {
                     continue;
                 }
                 if (!foundStopChar) {
-                    let extra = "";
+                    /** @type String[] */
+                    let extra = [];
                     if (isLastElemGeneric(query.elems, parserState)) {
-                        // @ts-expect-error
                         extra = [" after ", ">"];
                     } else if (prevIs(parserState, "\"")) {
                         throw ["Cannot have more than one element if you use quotes"];

--- a/src/librustdoc/html/static/js/search.js
+++ b/src/librustdoc/html/static/js/search.js
@@ -2319,7 +2319,7 @@ class DocSearch {
          * @param {rustdoc.ParserQueryElement} elem
          */
         function convertTypeFilterOnElem(elem) {
-            if (typeof elem.typeFilter === 'string') {
+            if (typeof elem.typeFilter === "string") {
                 let typeFilter = elem.typeFilter;
                 if (typeFilter === "const") {
                     typeFilter = "constant";

--- a/src/librustdoc/html/static/js/search.js
+++ b/src/librustdoc/html/static/js/search.js
@@ -1584,17 +1584,17 @@ class DocSearch {
      * This is effectively string interning, so that function matching can be
      * done more quickly. Two types with the same name but different item kinds
      * get the same ID.
-	 *
-	 * @template T extends string
-	 * @overload
+     *
+     * @template T extends string
+     * @overload
      * @param {T} name
      * @param {boolean=} isAssocType - True if this is an assoc type
      * @returns {T extends "" ? null : number}
-	 *
-	 * @param {string} name
+     *
+     * @param {string} name
      * @param {boolean=} isAssocType
      * @returns {number | null}
-	 * 
+     *
      */
     buildTypeMapIndex(name, isAssocType) {
         if (name === "" || name === null) {

